src/doc/trpl/advanced-linking.md

```diff
@@ -0,0 +1,175 @@
+you want to use your compiled project. Rust libraries are statically linked by
+default so you can use Rust-created binaries and libraries without installing
+the Rust runtime everywhere. By contrast, native libraries are dynamically
+linked by default, but sometimes it can be useful to change this.
+
+Linking is a very platform dependant topic - on some platforms, static linking
+may not be possible at all! This section assumes some basic familiarity with
+linking on your platform on choice.
+
+## Linux
+
+By default, all Rust programs on Linux will link to the system libc along with
+a number of other libraries Let's look at an example on a 64-bit linux machine
+with GCC and glibc (by far the most common libc on Linux):
+
+``` text
+$ cat example.rs
+fn main() {}
+$ rustc example.rs
+$ ldd example
+        linux-vdso.so.1 =>  (0x00007ffd565fd000)
+        libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007fa81889c000)
+        libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007fa81867e000)
+        librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x00007fa818475000)
+        libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007fa81825f000)
+        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007fa817e9a000)
+        /lib64/ld-linux-x86-64.so.2 (0x00007fa818cf9000)
+        libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007fa817b93000)
+```
+
+Dynamic linking on Linux can be undesirable if you wish to target older
+machines as applications compiled aginst newer versions glibc are not
+guaranteed to run against older versions.
+
+You can examine Rust linking arguments with an option to rustc. Newlines have
+been added for readability:
+
+``` text
+$ rustc example.rs -Z print-link-args
+"cc"
+    "-Wl,--as-needed"
+    "-m64"
+    [...]
+    "-o" "example"
+    "example.o"
+    "-Wl,--whole-archive" "-lmorestack" "-Wl,--no-whole-archive"
+    "-Wl,--gc-sections"
+    "-pie"
+    "-nodefaultlibs"
+    [...]
+    "-Wl,--whole-archive" "-Wl,-Bstatic"
+    "-Wl,--no-whole-archive" "-Wl,-Bdynamic"
+    "-ldl" "-lpthread" "-lrt" "-lgcc_s" "-lpthread" "-lc" "-lm" "-lcompiler-rt"
+```
+
+Arguments with a `-L` before them set up the linker search path and arguments
+ending with `.rlib` are linking Rust crates statically into your application.
+Neither of these are relevent for static linking so have been ommitted.
+
+The first step in being able to statically link is to obtain an object file.
+This can be achieved with `rustc --emit obj example.rs`, and creates a file
+called `example.o`, which you can see being passed in the command line above -
+rustc automatically deletes it when finished with it by default. As you now have
+the object file, you should be able to run the link command obtained with
+`print-link-args` to create perform the linking stage yourself.
+
+In order to statically link, there are a number of changes you must make. Below
```

Comment:
Yes, I've changed my mind about including this. It doesn't work any more anyway since rustc changed how link arguments are constructed :P 
I'll replace it all with musl.